### ISSUE TITLE
Utilities Failing to Import through Spring Tool Suite

### ISSUE THREAD

**Issue by jeyakatsa** [Consensys-Incorporated/teku#4437]
### Description
As a Core Developer Apprentice for the Ethereum Foundation, I'm using the Teku build as reference in order to build a prototype for the Altair Minimal Light Client (https://github.com/jeyakatsa/Altair----Minimal-Light-Client-Prototype). Yet, I'm having issues in regards to the utilities needed to be imported in order for the packages (and classes/interfaces within the build) to be error free...

### Steps to Reproduce (Bug)
1. Git Pull Teku folder into OS
2. Open Spring Tool Suite 4
3. Go to [File] -> [Open Projects From File System] and choose Teku folder from files

**Expected behavior:** 
No Errors packages, classes and interfaces

**Actual behavior:**
Errors on all packages, classes and interfaces.

**Image Reference**
![Teku Build Import Utility Issues](https://user-images.githubusercontent.com/51394348/135779259-f423f789-0c9d-43c2-a271-4488b4f1e7ca.jpg)

**Frequency:**
All the time

### Versions
* Software version: [`Unsure... Forked and Pulled the Teku Repo on Wednesday 9/29/21.`]
* Java version: [`jdk-15.0.1.9-hotspot`]
* OS Name & Version: [`Windows 10`]
* Virtual Machine software & version: [`Spring Tool Suite 4, Version: 4.7, Release: 2020`]

**Comment by rolfyone** [comment]
Thanks for raising. I'm assuming that you're importing the rest api definition?

There's been a couple of reports of specific issues due to automated importing of the rest api, and previously I've suggested importing the standard api which is written by hand...
The standard api can be found here. Admittedly 'teku specific' api's are missing from that definition, but all of the key api's are defined :
https://ethereum.github.io/beacon-APIs/

Last time I checked it was difficult to fix due to external dependencies of several libraries, I'll schedule some time to see if it's in a state to attempt to resolve yet...

**Comment by ajsutton** [comment]
I'm not at all familiar with spring tools suite but it looks like it's built on Eclipse.  It looks like Eclipse is having issues again because it doesn't separate test, test support and production code trees in the same way gradle does so it hits a dependency cycle where gradle and IntelliJ have no issues.

Eclipse isn't giving me very useful error messages to point out where it thinks the cycle is either unfortunately. We'll have to do some further analysis to work it out and then how to restructure things to make Eclipse happy.

**Comment by jeyakatsa** [comment]
I'll try to use Eclipse and/or Gradle... I'll get back if I still experience the same issues and close this issue if I resolve said issues.

Thanks for responding urgently you two.

**Comment by ajsutton** [comment]
I think the two places that Eclipse is having a hard time with is the `:protoarray` module tests depending on test fixtures from `:storage` and the `:data:serializer` module tests depending on test fixtures from `:data:provider`.

However it's still missing a number of testFixture dependencies - I suspect because of https://github.com/gradle/gradle/issues/18138 since the errors I'm seeing are from `eth-benchmark-tests` module. Not sure if we can resolve that or if it needs to wait on a gradle or eclipse fix.

**Comment by jeyakatsa** [comment]
@ajsutton downloading/installing/synchronizing Gradle with Eclipse quite nearly resolved the utility import issue entirely (which was a major roadblock towards moving onto the next phase of building)... From here I can at least begin building the skeleton needed for the Minimal Light Client prototype for the Altair fork (in Java). 

I'll close out this issue for now and open a new issue if I run into other major roadblocks.

Thanks a bunch you two!

-J
